### PR TITLE
added basic gitignore

### PR DESCRIPTION
**Pull request by JohnMurray**
Just some basic gitignore stuff that will eventually be requested anyways.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# sbt specific
+core/target/
+macros/target/
+project/project/
+project/target/
+tests/target/
```
Comment (densh) [inline .gitignore]: I think that just having "target/" rule will ignore all of those.

**Comment by densh** [comment]
Once target rules are collapsed into one it's good to go. Please squash it into one commit.

**Comment by JohnMurray** [comment]
@densh - collapsed rules for `target` and squashed.

**Comment by JohnMurray** [comment]
I lied... now it is squashed.

**Comment by densh** [comment]
Thanks!

**Pull request merged**